libcodechecker/analyze/log_parser.py

```diff
@@ -47,69 +119,28 @@ def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
         " " + sysroot + " - -v "
 
     LOG.debug("Retrieving default includes via '" + cmd + "'")
-    include_paths = []
-    try:
-        proc = subprocess.Popen(shlex.split(cmd),
-                                stdin=subprocess.PIPE,
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-
-        _, err = proc.communicate("")
-
-        do_append = False
-        for line in err.splitlines(True):
-            line = line.strip()
-            if line.startswith(end_mark):
-                do_append = False
-            if do_append:
-                # On OSX there are framework includes,
-                # where we need to strip the "(framework directory)" string.
-                # For instance:
-                # /System/Library/Frameworks (framework directory)
-                fpos = line.find("(framework directory)")
-                if fpos == -1:
-                    include_paths.append("-isystem " + line)
-                else:
-                    include_paths.append("-isystem " + line[0:fpos-1])
-
-            if line.startswith(start_mark):
-                do_append = True
-
-    except OSError as oerr:
-        LOG.error("Cannot find include paths: " + oerr.strerror + "\n")
-    return include_paths
+    err = get_compiler_err(cmd)
+    dump_compiler_info(output_path, "compiler_includes.json",
+                       {"compiler": compiler, "includes": err})
+    return parse_compiler_includes(err)
 
 
-def get_compiler_target(compiler):
+def get_compiler_target(compiler, output_path):
     """
     Returns the target triple of the given compiler as a string.
-
     """
-    target_label = "Target:"
-    target = ""
 
     cmd = compiler + ' -v'
     LOG.debug("Retrieving target platform information via '" + cmd + "'")
 
-    try:
-        proc = subprocess.Popen(shlex.split(cmd),
-                                stdin=subprocess.PIPE,
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-
-        _, err = proc.communicate("")
-        for line in err.splitlines(True):
-            line = line.strip().split()
-            if line[0] == target_label:
-                target = line[1]
-
-    except OSError as oerr:
-        LOG.error("Cannot find compiler target: " + oerr.strerror + "\n")
-
-    return target
+    err = get_compiler_err(cmd)
+    dump_compiler_info(output_path, "compiler_target.json",
```

Comment:
Does it make sense to both dump compiler info when getting the target and when getting the includes?